src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/TangibleAssetsNoteController.java

```diff
@@ -0,0 +1,106 @@
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.web.accounts.annotation.NextController;
+import uk.gov.companieshouse.web.accounts.annotation.PreviousController;
+import uk.gov.companieshouse.web.accounts.controller.BaseController;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.smallfull.notes.tangible.TangibleAssets;
+import uk.gov.companieshouse.web.accounts.service.company.CompanyService;
+import uk.gov.companieshouse.web.accounts.service.smallfull.TangibleAssetsNoteService;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+
+@Controller
+@NextController(ReviewController.class)
+@PreviousController(OtherAccountingPolicyController.class)
+@RequestMapping("/company/{companyNumber}/transaction/{transactionId}/company-accounts/{companyAccountsId}/small-full/note/tangible-assets")
+public class TangibleAssetsNoteController extends BaseController {
+
+    @Autowired
+    private TangibleAssetsNoteService tangibleAssetsNoteService;
+
+    @Autowired
+    private CompanyService companyService;
+
+    @Override
+    protected String getTemplateName() {
+        return "smallfull/tangibleAssetsNote";
+    }
+
+    @GetMapping
+    public String getTangibleAssetsNote(@PathVariable String companyNumber,
+        @PathVariable String transactionId,
+        @PathVariable String companyAccountsId,
+        Model model,
+        HttpServletRequest request) {
+
+        addBackPageAttributeToModel(model, companyNumber, transactionId, companyAccountsId);
+
+        try {
+            addDatesToModel(model, companyService.getCompanyProfile(companyNumber));
+            model.addAttribute("tangibleAssets", tangibleAssetsNoteService
+                .getTangibleAssets(transactionId, companyAccountsId, companyNumber));
+
+        } catch (ServiceException e) {
+            LOGGER.errorRequest(request, e.getMessage(), e);
+            return ERROR_VIEW;
+        }
+
+        return getTemplateName();
+    }
+
+    @PostMapping
+    public String postTangibleAssets(@PathVariable String companyNumber,
+        @PathVariable String transactionId,
+        @PathVariable String companyAccountsId,
+        @ModelAttribute("tangibleAssets") @Valid TangibleAssets tangibleAssets,
+        BindingResult bindingResult,
+        Model model,
+        HttpServletRequest request) {
+
+        addBackPageAttributeToModel(model, companyNumber, transactionId, companyAccountsId);
+
+        if (bindingResult.hasErrors()) {
+            return getTemplateName();
+        }
+
+        try {
+            addDatesToModel(model, companyService.getCompanyProfile(companyNumber));
```

Comment:
As discussed, I'd be inclined to add hidden form inputs which take the values of these dates. The dates can be added to the `TangibleAssets` model so that on form submission, the values bind to the model, removing the need for a repeat api call in the case where validation errors occur. 

If binding result errors occur at the moment, these values will be missing from the screen because they haven't been set by that point.